File: 0227-basic-calculator-ii/0227-basic-calculator-ii.cpp
class Solution {
public:
    int calculate(string s) {
        char sign = '+';
        int currentNum = 0, prevNum = 0;
        int ans = 0;
        
        for(int i = 0; i < s.length(); i++)
        {
            char currentChar = s[i];
            
            if(currentChar >= '0' && currentChar <= '9')
            {
                currentNum = currentNum * 10 + (currentChar - '0');
            }
            
            if(!(currentChar >= '0' && currentChar <= '9') && currentChar != ' ' || i == s.length() - 1)
            {
                if(sign == '+' || sign == '-')
                {
                    ans += prevNum;
                    prevNum = sign == '+' ? currentNum : -currentNum;
                }
                else if(sign == '*')
                {
                    prevNum = prevNum * currentNum;
                }
                else
                {
                    prevNum = prevNum / currentNum;
                }
                
                
                sign = currentChar;
                currentNum = 0;
            }
        }
        
        //3+2*2
        ans += prevNum;
        return ans;
    }
};

File: 0227-basic-calculator-ii/0227-basic-calculator-ii.py
class Solution:
    def calculate(self, s: str) -> int:
        stack = []
        
        curtNum = 0
        sign = "+"
        
        for i, c in enumerate(s):
            if c >= '0' and c <= '9':
                curtNum = curtNum * 10 + ord(c) - ord('0')
            
            
            if c in "+-*/" or i == len(s) - 1:
                
                if sign == "+":
                    stack.append(curtNum)
                elif sign == "-":
                    stack.append(-curtNum)
                    
                elif sign == "*":
                    stack[-1] = stack[-1] * curtNum
                else:
                    stack[-1] = int(stack[-1] / curtNum)
                
                sign = c
                curtNum = 0
                
        ans = 0
        
        while stack:
            ans += stack[-1]
            stack.pop()
        
        return ans